### ISSUE TITLE
Bug in cheffish 4.1.0 causing "wrong number of arguments" error

### ISSUE THREAD

**Issue by ramereth** [chef/chef-provisioning#567]
I'm using the latest chefdk-1.2.22 and was trying to using the fog provider with OpenStack with a known working config. When I ran it I got an error such as this:

```
    ArgumentError
    -------------
    wrong number of arguments (given 0, expected 1..2)
```

The stacktrace of which shows the following:

```
ArgumentError: machine[controller] (@recipe_files::/home/lance/git/osl/chef-repo/osuosl-cookbooks/osl-openstack/test/integration/chef-provisioning/controller.rb line 15) had an error: ArgumentError: wrong number of arguments (given 0, expected 1..2)
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/mixlib-config-2.2.4/lib/mixlib/config.rb:255:in `default'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/ohai-8.23.0/lib/ohai/mash.rb:179:in `from_hash'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/cheffish-4.1.0/lib/cheffish/merged_config.rb:6:in `block in initialize'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/cheffish-4.1.0/lib/cheffish/merged_config.rb:6:in `map'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/cheffish-4.1.0/lib/cheffish/merged_config.rb:6:in `initialize'
/home/lance/.chefdk/gem/ruby/2.3.0/gems/chef-provisioning-fog-0.20.0/lib/chef/provisioning/fog_driver/providers/openstack.rb:48:in `new'
/home/lance/.chefdk/gem/ruby/2.3.0/gems/chef-provisioning-fog-0.20.0/lib/chef/provisioning/fog_driver/providers/openstack.rb:48:in `compute_options_for'
/home/lance/.chefdk/gem/ruby/2.3.0/gems/chef-provisioning-fog-0.20.0/lib/chef/provisioning/fog_driver/driver.rb:142:in `canonicalize_url'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-provisioning-2.1.0/lib/chef/provisioning.rb:74:in `driver_for_url'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-provisioning-2.1.0/lib/chef/provisioning/chef_run_data.rb:105:in `driver_for_url'  
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-provisioning-2.1.0/lib/chef/provisioning/chef_run_data.rb:84:in `driver_for'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-provisioning-2.1.0/lib/chef/provider/machine.rb:108:in `new_driver'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-provisioning-2.1.0/lib/chef/provider/machine.rb:25:in `block in <class:Machine>'  
(eval):2:in `block in action_allocate'  
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/provider.rb:361:in `instance_eval'  
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/provider.rb:361:in `compile_and_converge_action'
(eval):2:in `action_allocate'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-provisioning-2.1.0/lib/chef/provider/machine.rb:38:in `block in <class:Machine>'
(eval):2:in `block in action_ready'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/provider.rb:361:in `instance_eval'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/provider.rb:361:in `compile_and_converge_action'  
(eval):2:in `action_ready'  
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-provisioning-2.1.0/lib/chef/provider/machine.rb:56:in `block in <class:Machine>'  
(eval):2:in `block in action_converge'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/provider.rb:361:in `instance_eval'  
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/provider.rb:361:in `compile_and_converge_action'
(eval):2:in `action_converge'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/provider.rb:145:in `run_action'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/resource.rb:622:in `run_action'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/runner.rb:69:in `run_action'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/runner.rb:97:in `block (2 levels) in converge'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/runner.rb:97:in `each'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/runner.rb:97:in `block in converge'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/resource_collection/resource_list.rb:94:in `block in execute_each_resource'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/resource_collection/stepable_iterator.rb:114:in `call_iterator_block'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/resource_collection/stepable_iterator.rb:103:in `iterate'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/resource_collection/resource_list.rb:92:in `execute_each_resource'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/runner.rb:96:in `converge'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/client.rb:670:in `block in converge'  
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/client.rb:665:in `catch'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/client.rb:665:in `converge'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/client.rb:704:in `converge_and_save'  
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/client.rb:284:in `run'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/application.rb:295:in `block in fork_chef_client'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/application.rb:283:in `fork'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/application.rb:283:in `fork_chef_client'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/application.rb:248:in `block in run_chef_client'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/local_mode.rb:44:in `with_server_connectivity'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/application.rb:236:in `run_chef_client'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/application/client.rb:464:in `sleep_then_run_chef_client'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/application/client.rb:451:in `block in interval_run_chef_client'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/application/client.rb:450:in `loop'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/application/client.rb:450:in `interval_run_chef_client'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/application/client.rb:434:in `run_application'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/lib/chef/application.rb:59:in `run'
/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.18.31/bin/chef-client:26:in `<top (required)>'
/opt/chefdk/bin/chef-client:57:in `load'
/opt/chefdk/bin/chef-client:57:in `<main>'
```

It appears that installing cheffish 4.1.1 seems to resolve the issue.

**Comment by tyler-ball** [comment]
Closing this on the same logic from https://github.com/chef/chef-provisioning-aws/pull/521#issuecomment-278421163

The next ChefDK will have the correct cheffish 4.1.1 included by default

**Comment by ramereth** [comment]
@tyler-ball looks like chefdk-1.3.40 pulls in cheffish-5.0.1 which still has the error. Does that version also include the bug/issue 4.0.0 had?

**Comment by tyler-ball** [comment]
@ramereth I just released Chef Provisioning 2.3.1 which includes another found Cheffish 5 issue. Would you try that and LMK if it solves your issue?

**Comment by ramereth** [comment]
@tyler-ball seems to be working now, thanks!
